doc/embedded/crosscompile_example.md

```diff
@@ -0,0 +1,132 @@
+
+fetch_mlpack()
+
+```
+
+#### Setting up include directories and source files
+
+The last part of our CMake file consists of merging all of the above components
+together. First we are going to find the OpenMP package if the user defined the
+variable above, then we will start including the directories of our
+program. You will need to do the same for your software if you are trying to
+integrate this example into an existing development, this should be done by
+defining an include variable using `set` directive or add them directly in
+`target_include_directories`.
+Regarding the source code, a similar process by either adding then to
+`add_executables` or by appending the source files to `SOURCES_FILES` variable. 
+
+Finally do not forget to add any external library that you need to link against
+in `target_link_libraries`.
+
+```cmake
+include_directories(BEFORE ${MLPACK_INCLUDE_DIRS})
+include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/src/)
+
+## User includes go here
+
+# Finally, add any cross-compilation support libraries (they may need to come
+# last).  If we are not cross-compiling, no changes will happen here.
+set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${CROSS_COMPILE_SUPPORT_LIBRARIES})
+
+## Add your source files to SOURCES_FILES list
+
+add_executable(RandomForest main.cpp ${SOURCES_FILES})
+target_sources(RandomForest PRIVATE ${SOURCE_FILES})
+
+## Do not forget to add the include variables in here.
+
+target_include_directories(RandomForest PRIVATE
+  ${MLPACK_INCLUDE_DIRS}
+  /path/to/your/own/include
+)
+
+## Do not forget to add libraries that you are linking against in here.
+
+target_link_libraries(RandomForest PRIVATE -static
+  ${MLPACK_LIBRARIES}
+)
+```
+
+#### Optimization and crosscompilation
+
+If you are interested in adding specific compiler flags to optimize operations
+on your hardware. Please feel free to look at mlpack
+`CMake/crosscompile-arch-flags.cmake` you can either integrate the variable by
+copy and paste them directly to this CMake file or you can
+copy the entire `crosscompile-arch-config.cmake` file from `mlpack/CMake` and included
+it close to `Autodownload.cmake` at start of this tutorial but in this case you need
+to specify the `ARCH_NAME` variable as we did in the this [tutorial](crosscompile_armv7.md).
+
+Once you have added all the source files and the headers for your applications,
+you can create your own build directory and build the software using `cmake`,
+and then `make`. Your cmake command should be similar to the following:
+
+```sh
+cmake \
+    -DBUILD_TESTS=ON \
+    -ARCH_NAME=(Check below) \
+    -DCMAKE_CROSSCOMPILING=ON \
+    -DCMAKE_TOOLCHAIN_FILE=../CMake/crosscompile-toolchain.cmake \
+    -DTOOLCHAIN_PREFIX=(Check below) \
+    -DCMAKE_SYSROOT=(Check below) \
+```
+
+In order to fill the `TOOLCHAIN_PREFIX` or the `CMAKE_SYSROOT`, or if
+you are interested in a different compiler toolchain please refer to the
+following [table](supported_boards.md).
```

Comment:
```suggestion
In order to fill the `TOOLCHAIN_PREFIX` or the `CMAKE_SYSROOT`, use
[this table](supported_boards.md).

If your preferred architecture is missing, or if the table needs an update,
please submit a PR to the repository and help us keep it up to date!
```

I dunno, maybe it is helpful to request that users help keep things up to date.  Let me know what you think :smile: